state/protocol/badger/mutator_test.go

```diff
@@ -123,6 +124,128 @@ func TestExtendValid(t *testing.T) {
 	})
 }
 
+func TestSealedIndex(t *testing.T) {
+	rootSnapshot := unittest.RootSnapshotFixture(participants)
+	util.RunWithFullProtocolState(t, rootSnapshot, func(db *badger.DB, state *protocol.MutableState) {
+		rootHeader, err := rootSnapshot.Head()
+		require.NoError(t, err)
+
+		// build a chain:
+		// G <- B1 <- B2 (resultB1) <- B3 <- B4 (resultB2, resultB3) <- B5 (sealB1) <- B6 (sealB2, sealB3) <- B7
+		// test that when B4 is finalized, can only find seal for G
+		// 					 when B5 is finalized, can find seal for B1
+		//					 when B7 is finalized, can find seals for B2, B3
+
+		// block 1
+		b1 := unittest.BlockWithParentFixture(rootHeader)
+		b1.SetPayload(flow.EmptyPayload())
+		err = state.Extend(context.Background(), b1)
+		require.NoError(t, err)
+
+		// block 2(result B1)
+		b1Receipt := unittest.ReceiptForBlockFixture(b1)
+		b2 := unittest.BlockWithParentFixture(b1.Header)
+		b2.SetPayload(flow.Payload{
+			Receipts: []*flow.ExecutionReceiptMeta{b1Receipt.Meta()},
+			Results:  []*flow.ExecutionResult{&b1Receipt.ExecutionResult},
+		})
```

Comment:
```suggestion
		b2.SetPayload(unittest.PayloadFixture(unittest.WithReceipts(b1Receipt)))
```
Noting the fixture helpers we have for this